.github/scripts/veristat-compare.py

```diff
@@ -38,165 +38,212 @@
-        if headers != HEADERS:
-            raise Exception(f'Unexpected table header for {filename}: {headers}')
-
-        for v in reader:
-            add = False
-            verdict = v[VERDICT_NEW]
-            diff = compute_diff(v)
-
-            if v[VERDICT_OLD] != v[VERDICT_NEW]:
-                changes = True
-                add = True
-                verdict = f'{v[VERDICT_OLD]} -> {v[VERDICT_NEW]}'
-                if v[VERDICT_NEW] == 'failure':
-                    new_failures = True
-                    verdict += ' (!!)'
-
-            if abs(diff * 100) > TRESHOLD_PCT:
-                changes = True
-                add = True
-
-            if not add:
-                continue
-
-            diff_txt = '{:+.1f} %'.format(diff * 100)
-            table.append([v[FILE], v[PROG], verdict, diff_txt])
-
-    return VeristatInfo(table=table,
-                        changes=changes,
-                        new_failures=new_failures)
-
-def format_table(headers, rows, html_mode):
-    def decorate(val, width):
-        s = str(val)
-        if html_mode:
-            s = s.replace(' -> ', ' &rarr; ');
-            s = s.replace(' (!!)', ' :bangbang: ');
-        return s.ljust(width)
-
-    column_widths = list(reduce(lambda acc, row: map(max, map(len, row), acc),
-                                rows,
-                                map(len, headers)))
-
-    with io.StringIO() as out:
-        def print_row(row):
-            out.write('| ')
-            out.write(' | '.join(map(decorate, row, column_widths)))
-            out.write(' |\n')
-
-        print_row(headers)
+    for record in reader:
+        add = False
+
+        verdict_old, verdict_new = (
+            record[VeristatFields.VERDICT_OLD],
+            record[VeristatFields.VERDICT_NEW],
+        )
+
+        if record[VeristatFields.VERDICT_DIFF] == "MISMATCH":
+            changes = True
+            add = True
+            verdict = f"{verdict_old} -> {verdict_new}"
+            if verdict_new == "failure":
+                new_failures = True
+                verdict += " (!!)"
+        else:
+            verdict = record[VeristatFields.VERDICT_NEW]
+
+        diff = get_state_diff(record[VeristatFields.TOTAL_STATES_DIFF])
+        if abs(diff) > TRESHOLD_PCT:
+            changes = True
+            add = True
+
+        if not add:
+            continue
+
+        table.append([
+            record[VeristatFields.FILE_NAME],
+            record[VeristatFields.PROG_NAME],
+            verdict,
+            f"{diff:+.2f} %"
+        ])
+
+    return VeristatInfo(
+        table=table,
+        changes=changes,
+        new_failures=new_failures
+    )
 
-        out.write('|')
-        out.write('|'.join(map(lambda w: '-' * (w + 2), column_widths)))
-        out.write('|\n')
 
-        for row in rows:
-            print_row(row)
+def html_decorate(input_str: str) -> str:
+    return input_str\
+        .replace(" -> ", " &rarr; ")\
```

Comment:
https://docs.python.org/3/library/html.html#html.escape ?

Reply:
It escapes html tags, but doesn't replace text with a nicer symbols

Reply:
> It escapes html tags, but doesn't replace text with a nicer symbols

ha yeah.... `->` would have been translated to `-&gt;` instead.